org.eclipse.dawnsci.analysis.dataset.test/src/org/eclipse/dawnsci/analysis/dataset/slicer/DynamicSliceViewIteratorTest.java

```diff
@@ -0,0 +1,124 @@
+package org.eclipse.dawnsci.analysis.dataset.slicer;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.eclipse.dawnsci.analysis.dataset.MockDynamicLazyDataset;
+import org.eclipse.january.DatasetException;
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.DatasetFactory;
+import org.eclipse.january.dataset.DatasetUtils;
+import org.eclipse.january.dataset.DoubleDataset;
+import org.eclipse.january.dataset.IDynamicDataset;
+import org.eclipse.january.dataset.IntegerDataset;
+import org.eclipse.january.dataset.Slice;
+import org.junit.Test;
+
+public class DynamicSliceViewIteratorTest {
+
+	@Test
+	public void testFail() {
+		
+		int[] fastest = {1,2,3,4,5,6};
+		
+		MockDynamicLazyDataset mock = getData(fastest);
+		MockDynamicLazyDataset mockkey = getKey(fastest);
+		MockDynamicLazyDataset mockfinished = getFinished();
+		
+		DynamicSliceViewIterator iterator = new DynamicSliceViewIterator(mock, new IDynamicDataset[]{mockkey}, mockfinished, 2);
+		iterator.setMaxTimeout(1000);
+		
+		boolean hasNext = iterator.hasNext();
+		
+		assertFalse(hasNext);
+	}
+	
+	@Test
+	public void testPass() throws DatasetException {
+		
+		int[] fastest = {1,2,3,4,5,6};
+		
+		MockDynamicLazyDataset mock = getData(fastest);
+		MockDynamicLazyDataset mockkey = getKey(fastest);
+		MockDynamicLazyDataset mockfinished = getFinished();
+		
+		mockkey.getParent().iadd(1);
+		mock.setAllowIncrement(true);
+		mockkey.setAllowIncrement(true);
+		
+		DynamicSliceViewIterator iterator = new DynamicSliceViewIterator(mock, new IDynamicDataset[]{mockkey}, mockfinished, 2);
+		iterator.setMaxTimeout(1000);
+		
+		for (int i = 0 ; i < fastest.length; i++) {
+			boolean hasNext = iterator.hasNext();
+			assertTrue(hasNext);
+			Dataset slice = DatasetUtils.convertToDataset(iterator.next().getSlice());
+			assertEquals(i*10*11, (double)slice.sum(),0.000001);
+		}
+	}
+	
+	@Test
+	public void testPassPartial() throws DatasetException {
+		
+		int[] fastest = {1,2,3,4,5,6};
+		int end = 3;
+		
+		MockDynamicLazyDataset mock = getData(fastest);
+		MockDynamicLazyDataset mockkey = getKey(fastest);
+		MockDynamicLazyDataset mockfinished = getFinished();
+		
+		mockkey.getParent().getSliceView(new Slice[]{new Slice(0,end,1)}).iadd(1);
+		mock.setAllowIncrement(true);
+		mockkey.setAllowIncrement(true);
+		
+		DynamicSliceViewIterator iterator = new DynamicSliceViewIterator(mock, new IDynamicDataset[]{mockkey}, mockfinished, 2);
+		iterator.setMaxTimeout(1000);
+		
+		for (int i = 0 ; i < end; i++) {
+			boolean hasNext = iterator.hasNext();
+			assertTrue(hasNext);
+			System.out.println(i);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)